core/healthcheck.py

```diff
@@ -1,8 +1,22 @@
 from django.http import JsonResponse
 from elasticsearch import Elasticsearch
+from django.conf import settings
 
 def elasticsearch_healthcheck(request):
-    client = Elasticsearch("http://localhost:9200")
-    if client.ping():
-        return JsonResponse({"status": "ok"})
-    return JsonResponse({"status": "error"}, status=500)
+    # Get Elasticsearch host from settings
+    es_config = getattr(settings, 'ELASTICSEARCH_DSL', {}).get('default', {})
+    hosts = es_config.get('hosts', 'http://localhost:9200')
+    
+    # Ensure hosts is a list for Elasticsearch client
+    if isinstance(hosts, str):
+        hosts = [f"http://{hosts}"]
```

Comment:
If the hosts string already contains "http://" or "https://", this will prepend another "http://", resulting in an invalid URL. Consider checking if the string already starts with "http://" or "https://" before prepending.